docs/concepts/api-extension/custom-resources.md

```diff
@@ -72,6 +223,318 @@ implementations for your custom resources by writing and deploying your own stan
 The main API server delegates requests to you for the custom resources that you handle,
 making them available to all of its clients.
 
+### Choosing which method to use to add a User-defined Resource
+
+CRDs is easier to use.  Aggregated APIs are more flexible.  Choose the one that best meets your needs.
+
+Typically, CRDs are a good fit if:
+
+
+
+*   You are have a handful of fields 
+*   You are using the resource within your company, or as part of a small open-source project (as opposed to a commercial product)
+
+<table>
+  <tr>
+   <td colspan="2" >
+<strong>Ease of use comparison</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>CRDs
+   </td>
+   <td>Aggregated API
+   </td>
+  </tr>
+  <tr>
+   <td>Do not require programming.
+<p>
+For writing a controller for the custom resources, user can chose any language for the controller.
+   </td>
+   <td>Requires programming in Go and building binary and image.
+<p>
+ 
+<p>
+For writing a controller for the custom resources, user can chose any language for the controller.
+   </td>
+  </tr>
+  <tr>
+   <td>No additional service to run; CRs are handled by API Server
+   </td>
+   <td>An additional service to create and that could fail.
+   </td>
+  </tr>
+  <tr>
+   <td>No ongoing support once the CRD is created.
+<p>
+Any bugfixes picked up as part of normal Kubernetes Master upgrades.
+   </td>
+   <td>May need to periodically pickup bugfixes from upstream and rebuild and update the Aggregated APIserver. 
+   </td>
+  </tr>
+  <tr>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td>You don't need to handle multiple versions of your API.<ul>
+
+<li>E.g. when you control the client for this resource, so you can upgrade it in sync with the API.</li></ul>
+
+   </td>
+   <td>You need to handle multiple versions of your API.<ul>
+
+<li>E.g. when developing an extension to share with the world.</li></ul>
+
+   </td>
+  </tr>
+</table>
+
+
+
+<table>
+  <tr>
+   <td colspan="4" ><strong>Flexibility Comparison</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>Feature
+   </td>
+   <td>What it does
+   </td>
+   <td>CRDs
+   </td>
+   <td>Aggregated API
+   </td>
+  </tr>
+  <tr>
+   <td>Validation
+   </td>
+   <td rowspan="2" >These help users prevent errors and allow you to evolve your API independently of your clients.  
+<p>
+These features are most useful when there are many clients, and they can't all update at the same time.
+   </td>
+   <td>Alpha feature of CRDs in v1.8.  Checks limited to what is supported by OpenAPI v3.0.
+   </td>
+   <td>Yes, arbitrary validation checks.
+   </td>
+  </tr>
+  <tr>
+   <td>Defaulting
+   </td>
+   <td>No.  But can get same effect with an Initializer. (requires programming).
```

Comment:
You would need a mutating admission webhook to truly get the same effect, since initializers don't run on updates.